default to `CppRuntime_Gcc` on Solaris

BarrOff opened the pull request:
Hello, 

I tried to compile druntime using DMD on a recent OpenIndiana machine. I ran into a problem, because on Solaris `CppRuntime_Sun` gets predefined in DMD. However, Developer Studio has been replaced by gcc in [OpenIndiana](https://www.openindiana.org/documentation/faq/#how-does-openindiana-differ-from-opensolaris), in [OmniOS also](https://github.com/omniosorg/omnios-build/blob/r151040/doc/ReleaseNotes.md) and I suppose in SmartOS too. The only Solaris system still using Developer Studio is the official Oracle Solaris, where Gcc is also available.

This patch would predefine `CppRuntime_Gcc` on Solaris systems, instead of `CppRuntime_Sun`. Some further discussion about this can be found [here](https://github.com/dlang/druntime/pull/3624#issuecomment-979449343)

Another possible solution could be to let the build script detect and define an `Illumos` version. One could insert a case for `Illumos` into the switch statement before the `Sun` case. I would need some help here, as I have not tinkered with the build scripts before. Appending this to the end of `osmodel.mak` might work:

```sh
ifeq ($(OS), solaris)
  ILLUMOS?=$(shell uname -a | grep -i illumos)
  ifneq (, $(ILLUMOS))
    DFLAGS+="-version=Illumos"
  endif
endif
```

I would really appreciate feedback and help with this.

BarrOff committed:
default to `CppRuntime_Gcc` on Solaris

ibuclaw commented on src/dmd/target.d:
```diff
@@ -1451,7 +1451,7 @@ struct TargetCPP
         case Gcc:         return predef("CppRuntime_Gcc");
         case Microsoft:   return predef("CppRuntime_Microsoft");
-        case Sun:         return predef("CppRuntime_Sun");
+        case Sun:         return predef("CppRuntime_Gcc");
```
You don't want to change this, rather make the Solaris runtime default to Gcc.

https://github.com/dlang/dmd/blob/28d77d18c79ad0b7b5fcdb8398c1fa0074df1f43/src/dmd/target.d#L1308-L1309

RazvanN7 commented:
ping @BarrOff

BarrOff committed:
Revert "default to `CppRuntime_Gcc` on Solaris"

This reverts commit 28d77d18c79ad0b7b5fcdb8398c1fa0074df1f43.

BarrOff committed:
make the Solaris runtime default to Gcc

BarrOff commented:
@ibuclaw @RazvanN7 

Sorry, I have been really busy the past days.
I reverted my previous commit and changed the runtime according to the suggestion.
